content/docs/reference/concepts.md

```diff
@@ -1,9 +1,9 @@
 ---
-title: Architecture & Concepts
+title: Pulumi Cloud Development
```

Comment:
I think keep this “Architecture and Concepts” for now - or something similar. 

Pulumi Cloud Development is too vague (could apply to any content in this site!). In fact, instead I think we want to rewrite the content in this page to use more consistent terminology (I am not sure the Pulumi Cloud Development Platfrom is actually an official name we use in general).